### ISSUE TITLE
Cannot use number as first letter in component of account name

### ISSUE THREAD

**Issue by blais** [beancount/beancount#301]
**[Original report](https://bitbucket.org/blais/beancount/issue/301) by arianaut (Bitbucket: [arianaut](https://bitbucket.org/arianaut)).**

----------------------------------------

https://docs.google.com/document/d/1wAMVrKIA2qtRGmoVDSUBJGmYZSygUaR0uOMW1GV3YE0/edit

The documentation says that 

Each component of the account names begin with a capital letter or a number 

However this does not seem to be correct (example error from web UI:

Invalid token: 'Expenses:4abc'

**Comment by blais** [comment]
**Original comment by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

This works for me with 2.1.2.  Which version of beancount do you have?

**Comment by blais** [comment]
**Original comment by arianaut (Bitbucket: [arianaut](https://bitbucket.org/arianaut)).**

----------------------------------------

Also 2.1.2, from Arch Linux AUR.

```
$ pac -Qs beancount
local/beancount 2.1.2-1
    Double-Entry Accounting from Text Files
local/fava 1.7-1
    Web interface for beancount
$ bean-check finances.org 
.../finances.org:85:      syntax error, unexpected COLON, expecting EOL or COMMENT
```
This is if line 85 is

```
2018-01-01 open Expenses:US:Hobbies:4ABC
```

If I change it to 

```
2018-01-01 open Expenses:4ABC
```

I get 

```
$ bean-check finances.org 
.../finances.org:85:      Invalid token: 'Expenses:4ABC'
```

**Comment by blais** [comment]
**Original comment by Zhuoyun Wei (Bitbucket: [wzyboy](https://bitbucket.org/wzyboy), GitHub: [wzyboy](https://github.com/wzyboy)).**

----------------------------------------

```
$ pacman -Qs beancount
local/beancount 2.1.2-1
    Double-Entry Accounting from Text Files
local/fava 1.7-1
    Web interface for beancount
$ cat 301.bean 
2018-01-01 open Expenses:US:Hobbies:4ABC
2018-02-01 open Expenses:4ABC
$ bean-check 301.bean 
$ bean-check --version
Beancount 2.1.2
```

I cannot reproduce the issue with AUR beancount 2.1.2.

**Comment by blais** [comment]
**Original comment by arianaut (Bitbucket: [arianaut](https://bitbucket.org/arianaut)).**

----------------------------------------

Oops, a copy of beancount 2.0 was shadowing the system installed version. `which bean-check` prints `/usr/bin/bean-check`, but the shadowed version sat on the Python import path.  This behavior is fixed as of 2.1.2
